js/analyzescorer.js

```diff
@@ -0,0 +1,168 @@
+            this.__score = this.__score.concat( subScore );
+        }
+    }
+    this.__totalScore = this.totalScore();
+};
+
+/**
+ * scoring function that returns results based on the resultobj from the analyzer matched with
+ * the scorearrays in the scoring config.
+ * @param obj
+ * @returns {{name: (analyzerScoring.scoreName), score: number, text: string}}
+ */
+YoastSEO_AnalyzeScorer.prototype.genericScore = function( obj ){
+    if( typeof obj !== "undefined" ) {
+        var scoreObj = this.scoreLookup( obj.test );
+        var score = { name: scoreObj.scoreName, score: 0, text: "" };
+        for ( var i = 0; i < scoreObj.scoreArray.length; i++ ) {
+            this.setMatcher( obj, scoreObj, i );
+            switch ( true ) {
+                case ( typeof scoreObj.scoreArray[i].type === "string" && this.result[scoreObj.scoreArray[i].type] ):
+                    return this.returnScore( score, scoreObj, i );
+                case ( typeof scoreObj.scoreArray[i].min === "undefined" && this.matcher <= scoreObj.scoreArray[i].max ):
+                    return this.returnScore( score, scoreObj, i );
+                case ( typeof scoreObj.scoreArray[i].max === "undefined" && this.matcher >= scoreObj.scoreArray[i].min ):
+                    return this.returnScore( score, scoreObj, i );
+                case ( this.matcher >= scoreObj.scoreArray[i].min && this.matcher <= scoreObj.scoreArray[i].max ):
+                    return this.returnScore( score, scoreObj, i );
+                default:
+                    break;
+            }
+        }
+        return score;
+    }
+};
+
+/**
+ * sets matcher and resultvariables so the scorefunction can use this.
+ * @param obj
+ * @param scoreObj
+ * @param i
+ */
+YoastSEO_AnalyzeScorer.prototype.setMatcher = function( obj, scoreObj, i ) {
+    this.matcher = parseFloat( obj.result );
+    this.result = obj.result;
+    if( typeof scoreObj.scoreArray[i].matcher !== "undefined" ){
+        this.matcher = parseFloat( this.result[scoreObj.scoreArray[i].matcher] );
+    }
+};
+
+/**
+ * finds the scoringobject by scorename for the current result.
+ * @param name
+ * @returns scoringObject
+ */
+YoastSEO_AnalyzeScorer.prototype.scoreLookup = function( name ) {
+    for ( var ii = 0; ii < this.scoring.length; ii++ ){
+        if ( name === this.scoring[ii].scoreName ){
+            return this.scoring[ii];
+        }
+    }
+};
+
+/**
+ * fills the score with score and text from the scoreArray and runs the textformatter.
+ * @param score
+ * @param scoreObj
+ * @param i
+ * @returns scoreObject
+ */
+YoastSEO_AnalyzeScorer.prototype.returnScore = function( score, scoreObj, i ) {
+    score.score = scoreObj.scoreArray[i].score;
+    score.text = this.scoreTextFormat( scoreObj.scoreArray[i], scoreObj.replaceArray );
+    return score;
+};
+
+/**
+ * Formats the resulttexts with variables. Uses a value, source, sourceObj or scoreObj for the replacement source
```

Comment:
I'm unsure what the difference between sourceObj and source is, I would think source is a property of sourceObj, but I'm not sure. Can you perhaps elaborate in the comments?